fastai/vision/gan.py

```diff
@@ -276,7 +338,11 @@ def show_results(x:InvisibleTensor, y:TensorImage, samples, outs, ctxs=None, max
     return ctxs
 
 # Cell
-def gan_loss_from_func(loss_gen, loss_crit, weights_gen=None):
+def gan_loss_from_func(
+    loss_gen:callable, # A loss function for the generator. It should take in the generator output images and target real images and provide an evaluation.
+    loss_crit:callable, # A loss function for the critic. It should take in the images, real or fake, and also labels indicating if it's real or fake, and provide an evaluation.
+    weights_gen:(None, list, tuple)=None # Weights for the generator and critic loss function
+):
```

Comment:
Isn't the `loss_crit` only passed labels and not any images?

Also rephrased to be more concise.
```suggestion
def gan_loss_from_func(
    loss_gen:callable, # A loss function for the generator. Evaluates generator output images and target real images
    loss_crit:callable, # A loss function for the critic. Evaluates predictions of real and fake images.
    weights_gen:(None, list, tuple)=None # Weights for the generator and critic loss function
):
```

Reply:
You are right, thanks for the correction!